src/pages/Search/SearchAddApproverPage.tsx

```diff
@@ -0,0 +1,214 @@
+    const [allReports] = useOnyx(ONYXKEYS.COLLECTION.REPORT, {canBeMissing: true});
+    const {clearSelectedTransactions, selectedReports} = useSearchContext();
+    const isSavingRef = useRef(false);
+
+    const currentUserDetails = useCurrentUserPersonalDetails();
+
+    // Get all possible approvers from all selected reports' policies
+    // An approver must be able to approve ALL selected reports
+    const allApprovers = useMemo(() => {
+        if (selectedReports.length === 0) {
+            return [];
+        }
+
+        const uniquePolicyIds = Array.from(new Set(selectedReports.map((selectedReport) => selectedReport.policyID)));
+        const employeeLists = uniquePolicyIds.map((policyID) => allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`]?.employeeList).filter((employeeList) => !!employeeList);
+        const firstWorkspaceEmployees = employeeLists.at(0);
+        const intersectedEmployees = firstWorkspaceEmployees ? lodashPick(firstWorkspaceEmployees, lodashIntersection(...employeeLists.map(Object.keys))) : {};
+        const policyMemberEmailsToAccountIDs = getMemberAccountIDsForWorkspace(intersectedEmployees, true, false);
+        return Object.values(intersectedEmployees)
+            .map((employee): SelectionListApprover | null => {
+                const isAdmin = employee?.role === CONST.REPORT.ROLE.ADMIN;
+                const email = employee?.email;
+
+                if (!email) {
+                    return null;
+                }
+                const accountID = Number(policyMemberEmailsToAccountIDs[email] ?? CONST.DEFAULT_NUMBER_ID);
+                const isPendingDelete = intersectedEmployees?.[accountID]?.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE;
+
+                if (isPendingDelete) {
+                    return null;
+                }
+
+                const isApproverOfAllReports = selectedReports.every((selectedReport) => {
+                    const report = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${selectedReport.reportID}`];
+
+                    if (!report) {
+                        return false;
+                    }
+
+                    return report.managerID === accountID;
+                });
+                if (isApproverOfAllReports) {
+                    return null;
+                }
+
+                const canApproveAllReports = selectedReports.every((selectedReport) => {
+                    const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${selectedReport.policyID}`];
+                    const report = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${selectedReport.reportID}`];
+
+                    if (!report || !policy) {
+                        return false;
+                    }
+
+                    if (report.managerID === accountID) {
+                        return true;
+                    }
+
+                    return isAllowedToApproveExpenseReport(report, accountID, policy);
```

Comment:
It seems like this mostly just checks if the employee is not the owner + some self approval settings check? Should we also be checking for membership on the policy?

Reply:
I'm confused about why we would ever add an approver who is outside the policy.